packages/dev/core/src/Animations/animatable.core.ts

```diff
@@ -397,7 +397,7 @@ export class Animatable {
      * Wait asynchronously for the animation to end
      * @returns a promise which will be fulfilled when the animation ends
      */
-    public waitAsync(): Promise<Animatable> {
+    public async waitAsync(): Promise<Animatable> {

```

Comment:
This function doesn't use `await`, so it doesn't need to be marked as `async`.

Reply:
There is a reasoning behind this which I wanted to discuss in an extra meeting, but I will revert this for now.
Semantically and technically there is no difference with or without async if the function directly returns a promise. There is a difference if the function returns a promise but does something beforehand.

Reply:
If there is more code than just returning a promise, then error handling has to be considered. If we think generally that we should make it `async` to handle this, then I'm okay with that.

Reply:
Correct. I left async only where we directly return a promise. If I missed one let me know and I'll revert it right after, because it's wrong.

Reply:
I hope you mean you "left async only where we **don't** directly return a promise" :)

This specific function creates the promise and returns it, so there is no reason for `async`, but I don't mind if we want to add `async` all the time just to be safe.

Reply:
I totally agree! this is one of my suggestions in the future linting meeting. I just don't want to break anything with this PR and then revert it because of some async keyword.